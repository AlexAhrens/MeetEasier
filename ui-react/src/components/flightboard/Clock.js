import React, { Component } from 'react';

class Clock extends Component {
  constructor(props) {
    super(props);
    this.state = {
      date: new Date()
    }
  }

  tick = () => {
    this.setState ({
      date: new Date()
    })
  }

  componentDidMount = () => {
    this.timerID = setInterval(
      () => this.tick(),
      1000
    );
  }

  componentWillUnmount = () => {
    clearInterval(this.timerID);
  }

  render() {
    return (
      <span id="clock">
        {this.state.date.toLocaleTimeString()}
      </span>
    );
  }
}

export default Clock;
