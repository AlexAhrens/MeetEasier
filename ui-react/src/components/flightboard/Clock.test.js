import Clock from './Clock';

describe('Flightboard Clock', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Clock />);
    const clock = wrapper.find('#clock');

    expect(clock).toBeDefined();
  });
});
