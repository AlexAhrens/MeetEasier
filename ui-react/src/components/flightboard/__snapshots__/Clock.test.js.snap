// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flightboard Clock renders correctly 1`] = `
<span
  id="clock"
>
  10:48:42 AM
</span>
`;
